src/java.base/share/native/libjava/io_util.c

```diff
@@ -142,32 +143,32 @@ writeSingle(JNIEnv *env, jobject this, jint byte, jboolean append, jfieldID fid)
     }
 }
 
-void
+int
 writeBytes(JNIEnv *env, jobject this, jbyteArray bytes,
            jint off, jint len, jboolean append, jfieldID fid)
 {
-    jint n;
+    jint nwritten = -1;
     char stackBuf[BUF_SIZE];
     char *buf = NULL;
     FD fd;
 
     if (IS_NULL(bytes)) {
         JNU_ThrowNullPointerException(env, NULL);
-        return;
+        return -1;
     }
 
     if (outOfBounds(env, off, len, bytes)) {
         JNU_ThrowByName(env, "java/lang/IndexOutOfBoundsException", NULL);
-        return;
+        return -1;
     }
 
     if (len == 0) {
-        return;
+        return 0;
     } else if (len > BUF_SIZE) {
-        buf = malloc(len);
+        buf = (char*)malloc(len * sizeof(char));
         if (buf == NULL) {
             JNU_ThrowOutOfMemoryError(env, NULL);
-            return;
+            return 0;
```

Comment:
why we are returning '0' here ? I can see, it is consistence with 'readBytes'  in case of OOM error we are returning 0 and other exceptions we are returning -1. Can you please help me understand if there is any specific reason for returning 0 in case of OOM ?.

Reply:
This a JNI function returning with a pending exception so the return value doesn't matter.